src/coins.h

```diff
@@ -188,14 +177,14 @@ struct CCoinsCacheEntry
     bool IsDirty() const noexcept { return m_flags & DIRTY; }
     bool IsFresh() const noexcept { return m_flags & FRESH; }
 
-    //! Only call Next when this entry is DIRTY, FRESH, or both
+    //! Only call Next when this entry is DIRTY or DIRTY-and-FRESH
     CoinsCachePair* Next() const noexcept
     {
         Assume(m_flags);
```

Comment:
I understand if you think it may modify too many things here, but I'd prefer an actual `Assume` instead of the comment:
```suggestion
    CoinsCachePair* Next() const noexcept
    {
        Assume(IsDirty());
        return m_next;
    }
```

(same for `Prev`)

Reply:
I don't see how that is different. `m_flags` being set implies `IsDirty()` now, so it is equivalent. I prefer not to change this.

Reply:
`Assume(m_flags);` would be true for fresh only - `Assume(IsDirty());` is stricter

Reply:
It's not possible to set fresh only now.

Reply:
Exactly, hence my recommendation